weave/derive_op.py

```diff
@@ -243,27 +241,19 @@ def resolve(**inputs):
                 or orig_op.name.endswith("run-history2")
             ):
                 wandb_api_ctx = wandb_api.get_wandb_api_context()
-                memo_ctx = memo._memo_storage.get()
 
                 def do_one(x):
                     if x == None or types.is_optional(first_arg.type):
                         return None
-                    memo_token = memo._memo_storage.set(memo_ctx)
-                    try:
-                        called = orig_op(x, **new_inputs)
-                        with wandb_api.wandb_api_context(wandb_api_ctx):
-                            with context.execution_client():
-                                # Use the use path to get caching.
-                                res = weave_internal.use(called)
-                                res = storage.deref(res)
-                    finally:
-                        memo._memo_storage.reset(memo_token)
+                    called = orig_op(x, **new_inputs)
+                    with wandb_api.wandb_api_context(wandb_api_ctx):
+                        with context.execution_client():
+                            # Use the use path to get caching.
+                            res = weave_internal.use(called)
+                            res = storage.deref(res)
                     return res
 
-                if USE_PARALLEL_DOWNLOAD:
-                    res = list(ThreadPoolExecutor(max_workers=10).map(do_one, list_))
-                else:
-                    res = list(map(do_one, list_))
+                res = list(parallelism.do_in_parallel(do_one, list_))
```

Comment:
we should still condition this on `USE_PARALLEL_DOWNLOAD`

Reply:
Sg, added back in